web/db/actions/User.ts

```diff
@@ -59,6 +73,50 @@ async function updateUserByUid(
   });
 }
 
+async function searchUser(
+  role?: Role,
+  type?: FosterType[],
+  size?: Size[],
+  preferredBreeds?: Breed[],
+  gender?: Gender[],
+  age?: Age[],
+  temperament?: Temperament[],
+  dogsNotGoodWith?: GoodWith[],
+  medical?: Medical[],
+  behavioral?: Behavioral[],
+  houseTrained?: Trained,
+  spayNeuterStatus?: Trained,
+  session?: ClientSession
+): Promise<IUser[] | null> {
```

Comment:
Take in an object with these keys instead. Session should be a separate parameter, however